src/v/base/format_to.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include <fmt/format.h>
+
+namespace fmt {
+
+using iterator = format_context::iterator;
+
+template<typename T>
+concept HasFormatToMethod = requires(const T& obj, iterator out) {
+    { obj.format_to(out) } -> std::same_as<iterator>;
+};
+
+/**
+ * A formatter that supports any type that implements a `format_to` method.
+ *
+ * For example:
+ *
+ * ```
+ * struct foo {
+ *   std::string a;
+ *   int32_t b;
+ *
+ *   fmt::iterator format_to(fmt::iterator it) const {
+ *     return fmt::print(it, "{{a: {}, b: {}}}", a, b)
```

Comment:
Should be `fmt::format_to` not `print` I think.

Reply:
oops yes, will send a fix